### PR TITLE
[feature-5119]: Change position standard texts per category

### PR DESCRIPTION
**Pull request by OscarBakker**
Add standardTextsField. Add util for position standard texts.

Ticket: [SIG-5119](https://gemeente-amsterdam.atlassian.net/browse/SIG-5119)

## Signalen

Before opening a pull request, please ensure:

- Make sure your PR title follows naming conventions: [feat-1234]: name feature
- Double-check your branch is based on `main` and targets `main`
- Pull request has tests (we are going for 100% coverage!)
- Code is well-commented, linted and follows project conventions
- Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: src/signals/settings/categories/types.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export interface CategoryFormValues {
   public_name: string
   use_calendar_days: number
   show_children_in_filter?: boolean
+  standard_texts?: any[]
```
Comment (OscarBakker) [inline src/signals/settings/categories/types.ts:14]: type

**Review by SireeKoolenWijkstra — COMMENTED** [review]
- [ ] When the arrow for changing the order is clicked, you get redirected to the previous screen immediately. I think a onSubmit is triggerd?
- [ ] When I press Annuleren, I get the pop up image I don't think this is part of the design.
- [ ] The menu is not conform design (is not grey for instance and the box is the same size as the component below). I like your version better. Is Linda onboard?

**Review by vdegraaf — COMMENTED** [review]
Functional:
- When I change the 1st text to the 2nd place, it immediately redirects back to the overview page. I suspect I have to press save for that. Same with the last one. But only when there are three texts

- When I don't change anything, it still patches and post the category and standard texts

**Comment by OscarBakker** [comment]
> * [ ]  When the arrow for changing the order is clicked, you get redirected to the previous screen immediately. I think a onSubmit is triggerd?
> * [ ]  When I press Annuleren, I get the pop up image I don't think this is part of the design.
> * [x]  The menu is not conform design (is not grey for instance and the box is the same size as the component below). I like your version better. Is Linda onboard?

1. fixed
2. is there for ages. I think it would be better to show our own modal here. Maybe for another story. ill Let Linda know
3. guess the design is more functional. Mine is just default and just like all the other selects.

File: src/shared/services/configuration/endpoint-definitions.js [hunk 2]
```diff
@@ -89,12 +89,17 @@ export default {
...
+  STANDARD_TEXTS_CATEGORY_ENDPOINT:
+    '/signals/v1/private/status-messages/category/',
+  // Private. Used by the incident management module to retrieve and query standard texts.
+  STANDARD_TEXTS_ENDPOINT: '/signals/v1/private/status-messages/',
+  // Private. Used by the incident management module to retrieve and search standard texts.
+  STANDARD_TEXTS_SEARCH_ENDPOINT: '/signals/v1/private/status-messages/search',
   // Private. Used by the incident management module to `GET` and `POST`
   // default response texts for incident handling. See
   // https://api.data.amsterdam.nl/api/swagger/?url=/signals/swagger/openapi.yaml#/default/get_signals_v1_public_terms_categories
   // for reference.
```
Comment (vdegraaf) [inline src/shared/services/configuration/endpoint-definitions.js:102]: I think the comments are slightly mixed up. The test here does not correspond with the TERMS_ENDPOINT

File: src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import type { StandardText } from 'types/api/standard-texts'
+
+import {
+  ButtonWrapper,
+  DefaultTextBody,
+  StyledButton,
+  TextWrapper,
+  Wrapper,
+} from './styled'
+import { useFetch } from '../../../../../hooks'
```
Comment (vdegraaf) [inline src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx:19]: the ../../../ can be removed. Same for /shared

File: src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    if (data?.results) {
+      const state = statusList.find(
+        (status) => status.value === selectedStatus
+      )?.key
+
+      const orderedStandardTexts = data.results.filter((result) => {
+        return result.state === state
+      })
+      setOrderedStandardTexts(orderedStandardTexts)
+    }
```
Comment (vdegraaf) [inline src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx:71]: It is not possible to assign StandardTexts to all statusses. Just this list `changeStatusOptionList`. Is defined in the same file.

File: src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  }, [data, selectedStatus])
+
+  const options = statusList.map((status) => {
+    return {
+      name: status.value,
+      value: status.value,
+      key: status.key,
+    }
+  })
```
Comment (vdegraaf) [inline src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx:80]: See comment above

File: src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        data-testid="standardTexts"
+        onChange={(e: FormEvent<HTMLSelectElement>) =>
+          setSelectedStatus(e.currentTarget.value)
+        }
+        options={options}
+        value={selectedStatus}
+      />
+
+      {orderedStandardTexts.map((standardText, index) => {
+        return (
```
Comment (vdegraaf) [inline src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx:96]: implicit return

File: src/signals/settings/categories/components/StandardTextsField/styled.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2023 Gemeente Amsterdam
+import { Button, themeColor } from '@amsterdam/asc-ui'
+import styled from 'styled-components'
+
+export const Wrapper = styled.div`
+  display: flex;
+  margin: 32px 0;
```
Comment (vdegraaf) [inline src/signals/settings/categories/components/StandardTextsField/styled.ts:8]: as discussed, the margin with the themeSpacing()

**Review by vdegraaf — APPROVED** [review]
Looks good!

Just one improvement would be nice. Something like "Geen standaard teksten beschikbaar" when there are none. But it's not in the design or ticket. 

Screenshot 2023-07-03 at 17 01 24

**Pull request merged**